src/optimum/rbln/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py

```
@@ -0,0 +1,601 @@
+                patch_size=config.patch_size,
+                temporal_patch_size=config.temporal_patch_size,
+                in_channels=config.in_channels,
+                embed_dim=config.hidden_size,
+            )
+        artifacts = torch.load(self.model_save_dir / self.subfolder / "torch_artifacts.pth", weights_only=False)
+        self.patch_embed.load_state_dict(artifacts["patch_embed"])
+
+    @classmethod
+    def save_torch_artifacts(
+        cls,
+        model: "Qwen2_5_VLForConditionalGeneration",
+        save_dir_path: Path,
+        subfolder: str,
+        rbln_config: RBLNQwen2_5_VisionTransformerPretrainedModelConfig,
+    ):
+        save_dict = {}
+        save_dict["patch_embed"] = model.patch_embed.state_dict()
+        torch.save(save_dict, save_dir_path / subfolder / "torch_artifacts.pth")
+
+    @classmethod
+    def wrap_model_if_needed(
+        cls, model: "PreTrainedModel", rbln_config: RBLNQwen2_5_VisionTransformerPretrainedModelConfig
+    ):
+        return Qwen2_5_VisionTransformerWrapper(model).eval()
+
+    def __getattr__(self, __name: str) -> Any:
+        def redirect(func):
+            return lambda *pargs, **kwargs: func(self, *pargs, **kwargs)
+
+        val = getattr(Qwen2_5_VisionTransformerPretrainedModel, __name)
+
+        if isinstance(val, Callable) and "self" in set(inspect.signature(val).parameters):
+            return redirect(val)
+        return val
+
+    @classmethod
+    def _update_rbln_config(
+        cls,
+        preprocessors: Union["AutoFeatureExtractor", "AutoProcessor", "AutoTokenizer"],
+        model: Optional["PreTrainedModel"] = None,
+        model_config: "PretrainedConfig" = None,
+        rbln_config: Optional[RBLNQwen2_5_VisionTransformerPretrainedModelConfig] = None,
+    ) -> RBLNQwen2_5_VisionTransformerPretrainedModelConfig:
+        if rbln_config.max_seq_lens is None:
+            rbln_config.max_seq_lens = [20480, 10240, 5120, 2048, 1024]
```

How were these values determined?

---

I agree that the hardcoded values for max_seq_lens were not appropriate. I have removed the default values and modified the code to raise an error if max_seq_lens is not provided.